### PR TITLE
Previously missed cppcheck performance fixes

### PR DESCRIPTION
**Pull request by tobbi**
This pull does (of course) not integrate the changes of all still open cppcheck perf fixes but only paths that have already been merged.

Please check thoroughly.

### CODE REVIEW

File: xbmc/music/tags/TagLoaderTagLib.cpp [hunk 1]
```diff
@@ -594,7 +594,7 @@ bool CTagLoaderTagLib::ParseXiphComment(Ogg::XiphComment *xiph, EmbeddedArt *art
     if (pictures[i].data().size())
     {
       string      mime =             pictures[i].mimeType().toCString();
-      if (mime.find("image/") != 0)
+      if (!mime.compare(0, 6, "image/"))
```
Comment (FernetMenta) [inline xbmc/music/tags/TagLoaderTagLib.cpp]: if string starts with "image/", find would return 0 and the expression is false
compare would also return 0 if string starts with "image/" and the ! make it true

**Comment by ace20022** [comment]
I'm pretty sure it won't compile, guess why.

**Comment by tobbi** [comment]
I'm on the rest of the issues. I finally managed to get it to recompile.

**Comment by tobbi** [comment]
I think I'm done now.

**Comment by FernetMenta** [comment]
looks ok to me. could you please squash the commits

**Comment by tobbi** [comment]
Done.

**Comment by FernetMenta** [comment]
jenkins builds this please

**Comment by FernetMenta** [comment]
typo :)
jenkins build this please

File: xbmc/LangInfo.cpp [hunk 2]
```diff
@@ -44,28 +44,25 @@ using namespace PVR;
 
 #define SPEED_UNIT_STRINGS 20200
 
-CLangInfo::CRegion::CRegion(const CRegion& region)
-{
-  m_strName=region.m_strName;
-  m_forceUnicodeFont=region.m_forceUnicodeFont;
```
Comment (anaconda) [inline xbmc/LangInfo.cpp]: `m_forceUnicodeFont` has disappeared here - see http://pastebin.com/NrZ30Xpu

**Pull request merged**